File: server/src/main/java/com/usforever/service/NoteService.java
package com.usforever.service;

import com.usforever.model.Note;

import java.util.List;

public interface NoteService {
    List<Note> getAllNotes();

    Note getNoteById(int id);

    Note createNote(Note newNote);

    Note updateNote(Note updatedNote);

    int deleteNote(int id);
}

File: client/src/service/NoteService.js
import axios from "axios";;

export default {

    getAllNotes() {
        return axios.get("/api/notes");
    },

    getNoteById(id) {
        return axios.get(`/api/notes/${id}`);
    },

    createNote(note) {
        return axios.post("/api/notes", note);
    },

    updateNote(id, note) {
        return axios.put(`/api/notes/${id}`, note);
    },
    deleteNote(id) {
        return axios.delete(`/api/notes/${id}`);
    }

}